libnetdata/ebpf/ebpf.c

```diff
@@ -507,20 +507,22 @@ void ebpf_update_module_using_config(ebpf_module_t *modules, struct config *cfg)
  * update the variables.
  *
  * @param em       the module structure
- * @param cfg      the configuration structure
- * @param cfg_file the filename to load
  */
-void ebpf_update_module(ebpf_module_t *em, struct config *cfg, char *cfg_file)
+void ebpf_update_module(ebpf_module_t *em)
 {
     char filename[FILENAME_MAX+1];
-    ebpf_mount_config_name(filename, FILENAME_MAX, ebpf_user_config_dir, cfg_file);
-    if (!ebpf_load_config(cfg, filename)) {
-        ebpf_mount_config_name(filename, FILENAME_MAX, ebpf_stock_config_dir, cfg_file);
-        if (!ebpf_load_config(cfg, filename)) {
-            error("Cannot load the ebpf configuration file %s", cfg_file);
+    int read_from_user = 1;
+    ebpf_mount_config_name(filename, FILENAME_MAX, ebpf_user_config_dir, em->config_file);
+    if (!ebpf_load_config(em->cfg, filename)) {
+        ebpf_mount_config_name(filename, FILENAME_MAX, ebpf_stock_config_dir, em->config_file);
+        read_from_user--;
+        if (!ebpf_load_config(em->cfg, filename)) {
+            error("Cannot load the ebpf configuration file %s", em->config_file);
             return;
         }
     }
 
-    ebpf_update_module_using_config(em, cfg);
+    // We only update when information is read from /etc/netdata/
+    if (read_from_user)
```

Comment:
Why do you need it at all? I don't think you should ignore stock configuration.

Reply:
I decided to add this condition, because I thought the following scenario:

-  User changes only `ebpf.d.conf`  to have `return` as default mode.
-  Whether the condition is not present, when we load the stock files, the `return` mode won't be applied. Thanks this condition we are using default values set inside `ebpf.d.conf`.

Reply:
I think this is incorrect. If you don't want any options from the stock file to be applied, comment them out.